### PR TITLE
Modernize IT plugins to use annotations

### PR DESCRIPTION
**Pull request by gnodet**
This is to be applied on top of #176.

### CODE REVIEW

**Comment by michael-o** [comment]
@gnodet Kindly rebase.

**Comment by gnodet** [comment]
> @gnodet Kindly rebase.

I plan to rebase on master once #176 is merged.

File: core-it-support/core-it-plugins/maven-it-plugin-fork/src/main/java/org/apache/maven/plugin/coreit/ForkGoalAggregatorMojo.java [hunk 1]
```diff
@@ -21,34 +21,36 @@
...
+    private List<MavenProject> reactorProjects;
 
     public void execute()
         throws MojoExecutionException
     {
-        for ( Object reactorProject : reactorProjects )
+        for ( MavenProject reactorProject : reactorProjects )
         {
-            MavenProject executedProject = ( (MavenProject) reactorProject ).getExecutionProject();
+            MavenProject executedProject = ( reactorProject ).getExecutionProject();
```
Comment (slawekjaranowski) [inline core-it-support/core-it-plugins/maven-it-plugin-fork/src/main/java/org/apache/maven/plugin/coreit/ForkGoalAggregatorMojo.java:53]: parentheses not needed

File: core-it-support/core-it-plugins/maven-it-plugin-no-default-comp/src/main/java/org/apache/maven/plugin/coreit/ItMojo.java [hunk 2]
```diff
@@ -32,25 +35,23 @@
...
      */
+    @Parameter( property = "touch.outputFile", defaultValue = "target/comp.properties" )
     private File outputFile;
 
     /**
      * NOTE: We don't specify a role hint here!
      *
-     * @component
      */
+    @org.apache.maven.plugins.annotations.Component
```
Comment (slawekjaranowski) [inline core-it-support/core-it-plugins/maven-it-plugin-no-default-comp/src/main/java/org/apache/maven/plugin/coreit/ItMojo.java:54]: import

File: core-it-support/core-it-plugins/maven-it-plugin-parameter-implementation/src/main/java/org/apache/maven/plugin/coreit/ParameterImplementationMojo.java [hunk 3]
```diff
@@ -30,6 +30,9 @@
 /**
  * Check that we correctly use the implementation parameter.
  *
+ * TODO: this mojo can not be converted to mojo annotations
+ * TODO: because the @Parameter#implementation has never been implemented.
+ *
  * @goal param-implementation
```
Comment (slawekjaranowski) [inline core-it-support/core-it-plugins/maven-it-plugin-parameter-implementation/src/main/java/org/apache/maven/plugin/coreit/ParameterImplementationMojo.java:36]: https://issues.apache.org/jira/browse/MPLUGIN-410

File: core-it-support/core-it-plugins/maven-it-plugin-active-collection/src/main/java/org/apache/maven/plugin/coreit/CheckMojo.java [hunk 4]
```diff
@@ -35,42 +38,35 @@
...
+    @Parameter( defaultValue = "${basedir}", readonly = true )
     private File basedir;
 
     /**
      * The available components, as a map.
-     *
-     * @component role="org.apache.maven.plugin.coreit.Component"
      */
-    private Map componentMap;
+    @org.apache.maven.plugins.annotations.Component
```
[the remark this answers is not in the document]
Comment (gnodet) [reply core-it-support/core-it-plugins/maven-it-plugin-active-collection/src/main/java/org/apache/maven/plugin/coreit/CheckMojo.java:57]: This is not doable.  The full declaration looks like:
```
    @org.apache.maven.plugins.annotations.Component
    private Map<String, Component> componentMap;
```
The `Component` interface is defined by at least 4 modules...

File: core-it-suite/pom.xml [hunk 5]
```diff
@@ -495,7 +832,7 @@ under the License.
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-dependency-plugin</artifactId>
-            <version>2.0</version>
+            <version>3.3.0</version>
```
Comment (slawekjaranowski) [inline core-it-suite/pom.xml:835]: Is version needed here, in line 358 you remove version tag

File: core-it-suite/src/test/resources/mng-3652/test-plugin/pom.xml [hunk 6]
```diff
@@ -10,24 +10,32 @@
...
+      <groupId>org.codehaus.plexus</groupId>
+      <artifactId>plexus-utils</artifactId>
+      <version>3.4.2</version>
     </dependency>
+<!--    <dependency>-->
+<!--      <groupId>junit</groupId>-->
+<!--      <artifactId>junit</artifactId>-->
+<!--      <version>3.8.1</version>-->
+<!--      <scope>test</scope>-->
+<!--    </dependency>-->
```
Comment (slawekjaranowski) [inline core-it-suite/src/test/resources/mng-3652/test-plugin/pom.xml:39]: please remove commented block

**Comment by gnodet** [comment]
@slawekjaranowski would you mind having another look ?

**Review by slawekjaranowski — APPROVED** [review]

**Comment by michael-o** [comment]
Looking through...

**Review by michael-o — CHANGES_REQUESTED** [review]
Massive!

File: core-it-support/core-it-plugins/maven-it-plugin-parameter-implementation/pom.xml [hunk 7]
```diff
@@ -42,7 +42,35 @@ under the License.
...
+      <scope>provided</scope>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-plugin-plugin</artifactId>
+        <version>3.6.5-20221006.213415-47</version>
```
[the remark this answers is not in the document]
Comment (gnodet) [reply core-it-support/core-it-plugins/maven-it-plugin-parameter-implementation/pom.xml:60]: Ah, right.  Though we're never releasing this project, so not sure it's as much a problem...
I'll try to get a release of `maven-plugin-tools` out asap.  
@michael-o @slawekjaranowski any problem with a release ?

[the remark this answers is not in the document]
Comment (slawekjaranowski) [reply core-it-support/core-it-plugins/maven-it-plugin-parameter-implementation/pom.xml:60]: @gnodet  - No problem with next releases, we have many resolved issues. 
Only open issues for next release should be reviewed.
Give me a day or two, I would like to look into it.

File: core-it-suite/src/test/resources/mng-3703/project/pom.xml [hunk 8]
```diff
@@ -19,7 +19,7 @@
       <plugins>
         <plugin>
           <artifactId>maven-site-plugin</artifactId>
-          <version>3.3</version>
+          <version>3.9.1</version>
```
[the remark this answers is not in the document]
Comment (gnodet) [reply core-it-suite/src/test/resources/mng-3703/project/pom.xml:22]: Moved to latest.

**Comment by gnodet** [comment]
Fwiw, this [commit](https://github.com/apache/maven-integration-testing/pull/184/commits/c4709de0e64a65a05c02db5c0ed2bb2ccf187a33) removes two ITs which [are not supported](https://issues.apache.org/jira/browse/MPLUGIN-435) by recent versions of the `maven-plugin-plugin`.

**Review by cstamas — APPROVED** [review]

**Review by michael-o — CHANGES_REQUESTED** [review]
Don't use `verifier.getSystemProperties().put(String, String)`, but `#AddCliArgument("-D....").

**Comment by gnodet** [comment]
> verifier.getSystemProperties()



> Don't use `verifier.getSystemProperties().put(String, String)`, but `#AddCliArgument("-D....")`.

Please consider the fact that I haven't changed much code, and I certainly have not introduced such constructs.  These changes are unrelated to the current PR which is already quite big.  I'd rather not expand the focus of this PR too much.  It's already been sitting there for 6 months and the longer it waits the harder it will be to review / maintain.

**Comment by michael-o** [comment]
> > verifier.getSystemProperties()
> 
> > Don't use `verifier.getSystemProperties().put(String, String)`, but `#AddCliArgument("-D....")`.
> 
> Please consider the fact that I haven't changed much code, and I certainly have not introduced such constructs. These changes are unrelated to the current PR which is already quite big. I'd rather not expand the focus of this PR too much. It's already been sitting there for 6 months and the longer it waits the harder it will be to review / maintain.

I agree, you are right. This is a separate topic.

**Comment by gnodet** [comment]
> > > verifier.getSystemProperties()
> > 
> > 
> > > Don't use `verifier.getSystemProperties().put(String, String)`, but `#AddCliArgument("-D....")`.
> > 
> > 
> > Please consider the fact that I haven't changed much code, and I certainly have not introduced such constructs. These changes are unrelated to the current PR which is already quite big. I'd rather not expand the focus of this PR too much. It's already been sitting there for 6 months and the longer it waits the harder it will be to review / maintain.
> 
> I agree, you are right. This is a separate topic.

Also, `verifier.getProperties()` isn't deprecated, so it may be worth deprecating it...

**Pull request merged**